extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -161,51 +315,58 @@ export abstract class AzureAuth implements vscode.Disposable {
 	 */
 	// TODO: Need to add resource we are getting the token for
 	// Can replace most refreshToken calls with getToken, it does the same thing
-	public async getToken(accountId: string, azureResource: azdata.AzureResource): Promise<AuthenticationResult> {
-
+	// TODO: separate getToken for MSAL
+	public async getTokenMsal(accountId: string, azureResource: azdata.AzureResource): Promise<AuthenticationResult> {
 		const cache = this.clientApplication.getTokenCache();
 		if (!cache) {
-			console.log('Error: Could not fetch token cache.');
+			Logger.error('Error: Could not fetch token cache.');
 		}
 		const resource = this.resources.find(s => s.azureResourceId === azureResource);
 		if (!resource) {
-			console.log('Error: Could not fetch the azure resource.');
+			Logger.error(`Error: Could not fetch the azure resource ${azureResource} `);
 		}
-		const accounts = await cache.getAllAccounts();
 		const account = await cache.getAccountByHomeId(accountId);
 		if (!account) {
-			console.log('Error: Could not fetch account when acquiring token');
+			Logger.error('Error: Could not fetch account when acquiring token');
 		}
 		let newScope = [`${resource.endpoint}/User.Read`];
 		// construct request
 		const tokenRequest = {
 			account: account,
 			scopes: newScope
 		};
-
 		try {
 			return await this.clientApplication.acquireTokenSilent(tokenRequest);
 		} catch (e) {
-			console.log('Failed to acquireTokenSilent');
-			console.log(e);
+			Logger.error('Failed to acquireTokenSilent', e);
 			// need to create an auth url request
 			const tenant: Tenant = {
 				id: account.tenantId,
 				displayName: ''
 			};
-			const authResult = await this.login(tenant, resource);
+			const authResult = await this.loginMsal(tenant, resource);
 			return authResult.response;
 		}
-		// Need: Account info
-		// Need: Scope
-		// Use account info to find correct account from cache
-		// Call clientApplication.getTokenSilent() in a try/catch
-		// In the catch if that fails, call clientApplication.getTokenInteractive
-		// return authResponse.accessToken
-	}
 
-	public async getTokenInteractive(authCodeRequest: AuthorizationCodeRequest): AuthenticationResult {
+	}
 
+	public async getToken(tenant: Tenant, resource: Resource, postData: AuthorizationCodePostData | TokenPostData | RefreshTokenPostData): Promise<OAuthTokenResponse> {
+		Logger.verbose('Fetching token');
+		const tokenUrl = `${this.loginEndpointUrl}${tenant.id}/oauth2/token`;
+		const response = await this.makePostRequest(tokenUrl, postData);
+		Logger.pii('Token: ', [{ name: 'access token', objOrArray: response.data }, { name: 'refresh token', objOrArray: response.data }],
```

Comment:
Why are you including the access token and refresh token multiple times? Seems like they should only be passed in one of the parameter arrays.

Reply:
the second array is passing in the actual strings to shorten

Reply:
Unless I'm very mistaken it takes the contents of both arrays and adds them to the message. The first array is ones it sanitizes (does some special logic, which includes shortening), and the second it just shortens. But all the strings end up in the final message. 

Did you verify this with enabling PII logging and looking at the output message yourself?

Reply:
I'll double check on this